openrouteservice/src/main/java/org/heigit/ors/fastisochrones/partitioning/InertialFlow.java

```diff
@@ -0,0 +1,333 @@
+package org.heigit.ors.fastisochrones.partitioning;
+
+import com.carrotsearch.hppc.IntArrayList;
+import com.carrotsearch.hppc.IntHashSet;
+import com.carrotsearch.hppc.cursors.IntCursor;
+import com.graphhopper.routing.util.EdgeFilter;
+import com.graphhopper.storage.Graph;
+import com.graphhopper.storage.GraphHopperStorage;
+import com.graphhopper.util.EdgeExplorer;
+import com.graphhopper.util.EdgeIterator;
+import org.heigit.ors.fastisochrones.partitioning.Projector.Projection;
+import org.heigit.ors.routing.graphhopper.extensions.edgefilters.EdgeFilterSequence;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.ExecutorService;
+
+import static org.heigit.ors.fastisochrones.partitioning.FastIsochroneParameters.*;
+
+/**
+ * Recursive implementation of InertialFlow algorithm for partitioning a graph.
+ *
+ * @author Hendrik Leuschner
+ */
+public class InertialFlow implements Runnable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(InertialFlow.class);
+    private static final int MIN_SPLITTING_ITERATION = 0;
+    private static final int MAX_SPLITTING_ITERATION = 268435456; //==2^28
```

Comment:
Yes, this is the maximum recursion depth. The number is chosen as an estimate. The average cell size is 50% of the "mother cell". If there is, by chance, a cell that, throughout 28 iterations, manages to get **on average** 64% of the nodes, it will still fit into this iteration maximum. Though the main reason to not have this be the full Integer Max value is to prevent buggy iterations that keep running on. So I think making this limit higher should be no problem.

Reply:
Set the limit to Integer.Max_Value, which is the limit that the id can have anyway, as the id is an integer.